### PR TITLE
Clean up rtf fixtures

### PR DESCRIPTION
**Pull request by Nicogene**
This PR increases the `RTF` required version to 1.3.2 and cleans up the rtf plugins after recent changes on devel branch of `RTF`.

Plugins tested, it can be merged.

### CODE REVIEW

File: src/rtf-plugins/fixture-managers/yarpmanager/CMakeLists.txt [hunk 1]
```diff
@@ -9,12 +9,12 @@ include_directories(SYSTEM ${RTF_INCLUDE_DIRS})
...
-add_library(rtf_fixturemanager_yarpmanager MODULE YarpFixManager.h
-                                                  YarpFixManager.cpp)
-set_property(TARGET rtf_fixturemanager_yarpmanager PROPERTY PREFIX "")
-set_property(TARGET rtf_fixturemanager_yarpmanager PROPERTY OUTPUT_NAME yarpmanager)
+rtf_add_plugin(rtf_fixturemanager_yarpmanager YarpFixManager.h
+                                              YarpFixManager.cpp
+                                              OUTPUT_NAME yarpmanager)
 
 target_link_libraries(rtf_fixturemanager_yarpmanager PRIVATE RTF::RTF
+                                                     PRIVATE RTF::RTF_dll
```
Comment (drdanz) [inline src/rtf-plugins/fixture-managers/yarpmanager/CMakeLists.txt]: The second private is not needed

File: src/rtf-plugins/fixture-managers/yarpplugin/CMakeLists.txt [hunk 2]
```diff
@@ -9,15 +9,15 @@ include_directories(SYSTEM ${RTF_INCLUDE_DIRS})
...
+rtf_add_plugin(rtf_fixturemanager_yarpplugin YarpPluginFixture.h
+                                             YarpPluginFixture.cpp
+                                             OUTPUT_NAME yarpplugin)
 
-target_link_libraries(rtf_fixturemanager_yarpplugin RTF::RTF
-                                                    YARP::YARP_OS
-                                                    YARP::YARP_init
-                                                    YARP::YARP_dev)
+target_link_libraries(rtf_fixturemanager_yarpplugin PRIVATE RTF::RTF
+                                                    PRIVATE RTF::RTF_dll
```
Comment (drdanz) [inline src/rtf-plugins/fixture-managers/yarpplugin/CMakeLists.txt]: same

File: src/rtf-plugins/fixture-managers/yarpmanager/CMakeLists.txt [hunk 3]
```diff
@@ -9,12 +9,12 @@ include_directories(SYSTEM ${RTF_INCLUDE_DIRS})
 
 add_definitions(-DSHLIBPP_FILTER_API)
 
-add_library(rtf_fixturemanager_yarpmanager MODULE YarpFixManager.h
-                                                  YarpFixManager.cpp)
-set_property(TARGET rtf_fixturemanager_yarpmanager PROPERTY PREFIX "")
-set_property(TARGET rtf_fixturemanager_yarpmanager PROPERTY OUTPUT_NAME yarpmanager)
+rtf_add_plugin(rtf_fixturemanager_yarpmanager YarpFixManager.h
+                                              YarpFixManager.cpp
+                                              OUTPUT_NAME yarpmanager)
```
Comment (drdanz) [inline src/rtf-plugins/fixture-managers/yarpmanager/CMakeLists.txt]: Didn't we add a `SOURCES` argument or something similar? I really don't like it in this way, the alignment is weird... 🤢

File: src/rtf-plugins/fixture-managers/yarpserver/CMakeLists.txt [hunk 4]
```diff
@@ -9,11 +9,11 @@ include_directories(SYSTEM ${RTF_INCLUDE_DIRS})
...
 add_definitions(-DSHLIBPP_FILTER_API)
 
-add_library(rtf_fixturemanager_yarpserver MODULE yarpserverfixture.cpp)
-set_property(TARGET rtf_fixturemanager_yarpserver PROPERTY PREFIX "")
-set_property(TARGET rtf_fixturemanager_yarpserver PROPERTY OUTPUT_NAME yarpserver)
+rtf_add_plugin(rtf_fixturemanager_yarpserver yarpserverfixture.cpp
+                                             OUTPUT_NAME yarpserver)
 
 target_link_libraries(rtf_fixturemanager_yarpserver PRIVATE RTF::RTF
+                                                    PRIVATE RTF::RTF_dll
```
Comment (drdanz) [inline src/rtf-plugins/fixture-managers/yarpserver/CMakeLists.txt]: same

**Comment by drdanz** [comment]
The changelog should clearly state that RTF 1.3.2 or later is required

File: src/rtf-plugins/fixture-managers/yarpmanager/CMakeLists.txt (hunk 3 above)
Comment (Nicogene) [reply src/rtf-plugins/fixture-managers/yarpmanager/CMakeLists.txt]: No there is not a `SOURCES` argument, they are taken as unparsed_arguments :disappointed: 
Do you think I have to change it in RTF? But in this way I have to bump another time the patch number.

**Comment by traversaro** [comment]
As the changelog is meant to be consumed for the next stable release, I would mention in it that the required RTF version is the 1.4.0 .

**Comment by drdanz** [comment]
Merged, thanks.

**Pull request merged**